### PR TITLE
Remove local storage and disable auto complete

### LINKED ISSUE #204: Disable browser from remembering password and email

### PR DESCRIPTION
**Pull request by harurang**
Resolves #204 and #202

## Purpose

- Remove form autocompletion.

- Use firebase instead of local storage to verify user is logged in in guard.

- Remove all cases of local storage and add user and organization to app store.

## Test

- [ ] Forms do not remember password or email unless user explicitly tells the browser it is okay.

### CODE REVIEW

File: src/app/effects/login.effects.ts [hunk 1]
```diff
@@ -6,28 +6,32 @@ import 'rxjs/add/operator/map';
...
-   * then store the results in localStorage.
+   * Listen for the LogIn action, log the afUser in, retrieve User,
+   * display success snackbar and navigate to settings page on success.
    * @type {Observable<any>}
    */
   @Effect()
   login$: Observable<Action> = this.actions
     .ofType(LoginActions.LOG_IN)
-    .map((action: LoginActions.LogIn) => action.payload)
+    .map((action: AppActions.LoadData) => action.payload)
```
Comment (tlmader) [inline src/app/effects/login.effects.ts]: This is a typo.

File: src/app/models/user.ts [hunk 2]
```diff
@@ -47,8 +47,7 @@ export const testLoginCredentials: any = [{
 }];
 
 /**
- * Represents the model in which users are stored in the firebase database.
- * @type {{id: string; organizationId: string; firstName: string; lastName: string}[]}
+ * Represents the model in which users are stored in the Firebase database.
  */
 export const testFbUsers: any[] = [
```
Comment (tlmader) [inline src/app/models/user.ts]: Add typing for `firebase.User`.

File: src/app/app.component.ts [hunk 3]
```diff
@@ -22,6 +21,7 @@ export class AppComponent implements OnInit, OnDestroy {
   appSubscription: Subscription;
   headerOptions: HeaderOptions;
   sidenavOptions: SidenavOptions[];
+  user: any;
```
Comment (tlmader) [inline src/app/app.component.ts]: Add typing.

Comment (harurang) [reply src/app/app.component.ts]: Added

**Comment by harurang** [comment]
The following changes have been made:

- FbUser typed

- User typed

- Unsubscribing from AppSubscription 

- Mutating copy of User object instead of reference

**Pull request merged**